src/transformers/models/seamless_m4t/feature_extraction_seamless_m4t.py

```diff
@@ -0,0 +1,293 @@
+    """
+
+    model_input_names = ["input_features", "attention_mask"]
+
+    def __init__(
+        self,
+        feature_size=80,
+        sampling_rate=16000,
+        num_mel_bins=80,
+        padding_value=0.0,
+        normalize_means=True,
+        normalize_vars=True,
+        stride=2,  # TODO: add to docstrings
+        lang_start_idx=256001,  # TODO: add to docstrings
+        src_lang="eng",
+        tgt_lang="fra",
+        language_code: Optional[List] = None,
+        **kwargs,
+    ):
+        self.num_mel_bins = num_mel_bins
+        self.normalize_means = normalize_means
+        self.normalize_vars = normalize_vars
+        self.return_attention_mask = True
+        self.stride = stride
+        self.lang_start_idx = lang_start_idx
+
+        language_code = language_code if language_code is not None else LARGE_SEAMLESS_M4T_LANGUAGE_CODES
+        language_code = [f"__{code}__" for code in language_code if "__" not in code]
+        self.lang_code_to_id = {code: lang_start_idx + i for i, code in enumerate(language_code)}
+
+        self.t2u_language_code = UNIT_SUPPORTED_LANGUAGES
+        self.t2u_lang_code_to_id = {code: i for i, code in enumerate(self.t2u_language_code)}
+
+        self.vocoder_language_code = VOCODER_SUPPORTED_LANGUAGES
+        self.vocoder_lang_code_to_id = {code: i for i, code in enumerate(self.vocoder_language_code)}
+
+        self._src_lang = f"__{src_lang}__"
+        self._tgt_lang = f"__{tgt_lang}__"
+
+        super().__init__(feature_size=feature_size, sampling_rate=sampling_rate, padding_value=padding_value, **kwargs)
+
+    @property
+    # Copied from transformers.models.nllb.tokenization_nllb.NllbTokenizer.src_lang
+    def src_lang(self) -> str:
+        return self._src_lang
+
+    @src_lang.setter
+    def src_lang(self, new_src_lang: str) -> None:
+        if "__" not in new_src_lang:
+            self._src_lang = f"__{new_src_lang}__"
+        else:
+            self._src_lang = new_src_lang
+
+    @property
+    def tgt_lang(self) -> str:
+        return self._tgt_lang
+
+    @tgt_lang.setter
+    def tgt_lang(self, new_tgt_lang: str) -> None:
+        if "__" not in new_tgt_lang:
+            self._tgt_lang = f"__{new_tgt_lang}__"
+        else:
+            self._tgt_lang = new_tgt_lang
+
+    def _extract_fbank_features(
+        self,
+        waveform: np.ndarray,
+    ) -> np.ndarray:
+        """
+        Get mel-filter bank features using TorchAudio. Note that TorchAudio requires 16-bit signed integers as inputs
+        and hence the waveform should not be normalized before feature extraction.
+        """
+        waveform = waveform * (2**15)  # Kaldi compliance: 16-bit signed integers
+        waveform = torch.from_numpy(waveform).unsqueeze(0) if len(waveform.shape) == 1 else torch.from_numpy(waveform)
+        features = ta_kaldi.fbank(waveform, num_mel_bins=self.num_mel_bins, sample_frequency=self.sampling_rate)
```

Comment:
Can we not use our implementation from `audio_utils`? This currently requires a hard dependency on torchaudio